hyperspace/cosmos/src/chain.rs

```diff
@@ -27,14 +30,35 @@ where
 		todo!()
 	}
 
-	async fn estimate_weight(&self, _messages: Vec<Any>) -> Result<u64, Self::Error> {
-		todo!()
+	async fn estimate_weight(&self, messages: Vec<Any>) -> Result<u64, Self::Error> {
+		Ok(0)
 	}
 
 	async fn finality_notifications(
 		&self,
 	) -> Pin<Box<dyn Stream<Item = <Self as IbcProvider>::FinalityEvent> + Send + Sync>> {
-		todo!()
+		let (ws_client, ws_driver) = WebSocketClient::new(self.websocket_url.clone())
+			.await
+			.map_err(|e| Error::from(format!("Web Socket Client Error {:?}", e)))
+			.unwrap();
```

Comment:
do we want to unwrap here?

Reply:
yeah, if the we can't connect to the websocket to receive finality notifications, we might as well just panic.